src/sentry/workflow_engine/processors/action.py

```diff
@@ -84,12 +87,17 @@ def update_workflow_fire_histories(
 
 # TODO(cathy): only reinforce workflow frequency for certain issue types
 def filter_recently_fired_workflow_actions(
-    filtered_action_groups: set[DataConditionGroup], event_data: WorkflowEventData
-) -> BaseQuerySet[Action]:
+    filtered_action_groups: dict[DataConditionGroup, int], event_data: WorkflowEventData
+) -> set[tuple[Action, int]]:
```

Comment:
🤔 not sure i agree with this method signature change - it seems like we're just grabbing the workflow id that we probably already have when we're invoking this method (we would need the workflow to get the list of dcgs to filter on).

Is there somewhere else we should be adding the workflow id instead? could we keep it super simple and get the `id` from process workflows when we're triggering actions? (or handle this in action.trigger to find the corresponding workflow so it's more generic)